src/include/pgagroal.h

```diff
@@ -228,6 +228,30 @@ struct prometheus_connection
    atomic_ullong query_count;           /**< The number of queries per connection */
 } __attribute__ ((aligned (64)));
 
+/**
+ * Max size of the cache (in bytes).
+ * If the cache request exceeds this size
+ * the caching should be aborted in some way.
+ */
+#define PROMETHEUS_MAX_CACHE_SIZE 8192
+
+/**
+ * A structure to handle the Prometheus response
+ * so that it is possible to serve the very same
+ * response over and over depending on the cache
+ * settings.
+ *
+ * The `valid_until` field stores the result
+ * of `time(2)` but instead of using a `time_t`
+ * there is an `atomic_ullong` to allow for
+ * concurrency management.
+ */
+struct prometheus_cache
+{
+   char data[PROMETHEUS_MAX_CACHE_SIZE];  /**< the payload */
```

Comment:
This should be a dynamic value - e.g. `char*`

Reply:
Forgive me if this is trivial, but I've a doubt. If we change the sized array to a `char *` it means we are going to `malloc` and `free` the pointer as the cache content grows and shrinks.
But such pointer is `mmap`ed into the `struct prometheus`, so if we don't allocate memory before the `mmap` call, processes will only share the pointer, not the content of the cache.
And `metrics_page()` is served by a `fork`ed process.
Doesn't this mean that `char *` needs special handling in this case?

Reply:
It is in a shared memory segment, so it will only be allocated once - and `memset` when we need it.

Reply:
I'm not so sure, for example when the configuration changes and the `pgagroal-cli reload` imposes a grow on the cache size, the `struct prometheus` is not going to be reallocated, so the inner carhe is not too, but it has to grow.
However, I will start implementing it and see when and where we are facing troubles.
Surely this is a little more than "caching to a string"!